src/server/lock_manager/mod.rs

```diff
@@ -175,11 +204,9 @@ impl LockMgr for LockManager {
             self.waiter_mgr_scheduler
                 .wake_up(lock_ts, hashes, commit_ts);
         }
-        // If these locks belong to a pessimistic transaction, clean up its wait-for entries
-        // in the deadlock detector.
-        //
-        // TODO: only clean up if the transaction once waited for locks.
-        if is_pessimistic_txn {
+        // If a pessimistic transaction is committed or rolled back and it once detected deadlock,
+        // clean up its wait-for entries in the deadlock detector.
+        if is_pessimistic_txn && self.remove_from_detected(lock_ts) {
```

Comment:
It seems that command `CleanUp` won't be persistent in Raft-Log? So is it more time-consuming than the hash table with `Lock`?